src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/respondentsreport/RespondentsReport.java

```diff
@@ -0,0 +1,127 @@
+        var respondentsReportDetailList = new ArrayList<RespondentsReportDetail>();
+        for (RespondentsReportSubmitEvent submitEvent : submitEvents) {
+            var caseData = submitEvent.getCaseData();
+            if (hasMultipleRespondents(caseData)) {
+                RespondentsReportDetail detail = new RespondentsReportDetail();
+                detail.setCaseNumber(caseData.getEthosCaseReference());
+                List<RespondentFields> fields = new ArrayList<>();
+                for (RespondentSumTypeItem r : caseData.getRespondentCollection()) {
+                    RespondentFields f = new RespondentFields();
+                    f.setRespondentName(r.getValue().getRespondentName());
+                    var rep = getRepresentative(r.getValue().getRespondentName(), caseData);
+                    f.setRepresentativeName(rep);
+                    f.setRepresentativeHasMoreThanOneRespondent(
+                        isRepresentativeRepresentingMoreThanOneRespondent(rep, caseData) ? "Y" : "N");
+                    fields.add(f);
+                }
+                detail.setRespondentDataList(fields);
+                respondentsReportDetailList.add(detail);
+            }
+        }
+        return respondentsReportDetailList;
+    }
+
+    private boolean hasMultipleRespondents(RespondentsReportCaseData caseData) {
+        if (CollectionUtils.isNotEmpty(caseData.getRespondentCollection())
+                && caseData.getRespondentCollection().size() > 1) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean isRepresentativeRepresentingMoreThanOneRespondent(String rep, RespondentsReportCaseData caseData) {
+        var count = 0;
+        if (CollectionUtils.isNotEmpty(caseData.getRepCollection())) {
+            for (RepresentedTypeRItem repItem : caseData.getRepCollection()) {
+                if (repItem.getValue().getNameOfRepresentative().equals(rep)) {
+                    for (RespondentSumTypeItem respItem : caseData.getRespondentCollection()) {
+                        if (respItem.getValue().getRespondentName().equals(repItem.getValue().getRespRepName())) {
+                            count = count + 1;
+                        }
+                    }
+                }
+            }
+        }
+        if (count > 1) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private String getRepresentative(String respName, RespondentsReportCaseData caseData) {
+        if (CollectionUtils.isNotEmpty(caseData.getRepCollection())) {
+            Optional<RepresentedTypeRItem> rep = caseData.getRepCollection().stream()
+                    .filter(a -> a.getValue().getRespRepName().equals(respName)).findFirst();
+            if (rep.isPresent()) {
+                return rep.get().getValue().getNameOfRepresentative();
+            } else {
```

Comment:
When you read the code the two return "N/A" statements stand out. You don't need the first one in the else statement however.

Reply:
yes, removed first "N/A"